pkgs/test/lib/src/runner/browser/browser_manager.dart

```diff
@@ -127,11 +137,14 @@ class BrowserManager {
       completer.completeError(error, stackTrace);
     });
 
-    return completer.future.timeout(Duration(seconds: 45), onTimeout: () {
+    return completer.future.timeout(Duration(seconds: 30), onTimeout: () {
       browser.close();
-      throw ApplicationException(
-          'Timed out waiting for ${runtime.name} to connect.\n'
-          'Browser output: ${utf8.decode(browser.output)}');
+      if (attempt >= _maxRetries) {
+        throw ApplicationException(
+            'Timed out waiting for ${runtime.name} to connect.\n'
+            'Browser output: ${utf8.decode(browser.output)}');
+      }
+      return _start(runtime, url, future, settings, configuration, attempt++);
```

Comment:
```suggestion
      return _start(runtime, url, future, settings, configuration, ++attempt);
```

(this would retry infinitely as written)

Reply:
😳

Reply:
Fixed